src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3644,16 +3644,41 @@ void addToStringMethod(MethodSymbol printMethod)
 
                 if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingToStringMethod))
                 {
-                    var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
-                    members.Add(toStringMethod);
+                    var baseToStringMethod = getBaseToStringMethod();
+                    if (baseToStringMethod is null || !baseToStringMethod.IsSealed)
+                    {
+                        var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
+                        members.Add(toStringMethod);
+                    }
                 }
                 else
                 {
                     var toStringMethod = (MethodSymbol)existingToStringMethod;
                     if (!SynthesizedRecordObjectMethod.VerifyOverridesMethodFromObject(toStringMethod, SpecialType.System_String, diagnostics) && toStringMethod.IsSealed && !IsSealed)
                     {
-                        diagnostics.Add(ErrorCode.ERR_SealedAPIInRecord, toStringMethod.Locations[0], toStringMethod);
+                        MessageID.IDS_FeatureSealedRecordToString.CheckFeatureAvailability(
+                            diagnostics,
+                            this.DeclaringCompilation,
+                            toStringMethod.Locations[0]);
+                    }
+                }
+
+                MethodSymbol? getBaseToStringMethod()
+                {
+                    var tmp = this.BaseTypeNoUseSiteDiagnostics;
+                    while (tmp is not null)
+                    {
+                        var method = tmp.GetSimpleNonTypeMembers(WellKnownMemberNames.ObjectToString)
+                            .OfType<MethodSymbol>()
+                            .SingleOrDefault(m => m.ParameterCount == 0 && m.ReturnType.SpecialType == SpecialType.System_String);
+
+                        if (method is not null)
+                            return method;
```

Comment:
>return method; [](start = 28, length = 14)

The fact that the name and the signature match doesn't mean this is an override of Object.ToString. We should be able to come up with a test scenario with base like that. If not defined in source, then in IL.

Reply:
I think we should mimic the check that is performed by ```SynthesizedRecordObjectMethod.VerifyOverridesMethodFromObject```, but I would also simplify its implementation by switching to GetSpecialTypeMember helper to locate the Object member and comparing it with ```leastOverridden``` instead of performing manual signature comparison.

---
In reply to: [601500624](https://github.com/dotnet/roslyn/pull/52029#discussion_r601500624) [](ancestors = 601500624)

Reply:
FYI, I just merged refactoring for ```SynthesizedRecordObjectMethod.VerifyOverridesMethodFromObject``` to take advantage of ```SpecialMemeber```.

---
In reply to: [601544539](https://github.com/dotnet/roslyn/pull/52029#discussion_r601544539) [](ancestors = 601544539,601500624)

Reply:
@AlekseyTs I made some changes to `getBaseToStringMethod` (compared least overriden to object.ToString(), replaced Linq usage with a loop). Is this what you had in mind?